### DISCUSSION TITLE
Vercel CLI deploys return 404s?

### DISCUSSION THREAD

**Discussion by osdiab** [vercel/vercel#5444] [Help] [answered]
Hello, I am trying to use the Vercel CLI to deploy my NextJS app alongside my (Heroku) API deploys in CI, instead of relying on the git automatic deploys. When a deploy is triggered through Git, Vercel deploys seem to work fine - but when I deploy using the `vercel --prod` command on my machine, it's resulting in a generic Vercel 404 page showing up when I go to the deployment domain.

The docs on the CLI are very light on details of what exactly differs between deploys created using the Vercel CLI vs deploys made via Git. For example when I run the `vercel` CLI, I noticed that the build step doesn't seem to happen, the command ends quickly and the Deployments page on the Vercel dashboard doesn't show any logs. So what exactly is the `vercel` CLI doing when it deploys?

Some details:

- My app is a monorepo, so the root directory is not at the root of the git repo
- Vercel CLI version is 20.1.4
- we want this behavior since we're an early stage company releasing frequent breaking changes to our API that need matching frontend changes, and we want to minimize the time that users might be served frontend and API changes that conflict with one another - hence simultaneous deploys as much as we can
- moving our API from a Node server in Heroku to Vercel is a possibility but beyond the scope of our current frontend migration from a static CRA app to NextJS/Vercel, hoping to keep them separate for the time being.

**Comment by mcsdevv** [comment]
Hello @osdiab, have you checked the output of the deployment in the dashboard?

**Comment by osdiab** [comment]
@mcsdevv yep, the output of the deployment just says there are no logs, and under Repository it says there's no branch. Does the `vercel` CLI command not actually run the build based on either what's locally on your machine nor whats in your git history? Super confused by this behavior.

**Reply by mcsdevv** [reply]
What do you find in the build outputs?

**Reply by osdiab** [reply]
There appears to be my source files but no build output visible (just loading forever)

![3248D5C2-B200-4ED0-B1FA-E53D40FDE208](https://user-images.githubusercontent.com/1890926/99812157-72b65d00-2b89-11eb-8036-982090e02b3a.jpeg)
![D53733EB-0E8E-4869-B209-0B6FD78E9A40](https://user-images.githubusercontent.com/1890926/99812166-74802080-2b89-11eb-955c-1dd5a85eee46.jpeg)

**Reply by osdiab** [reply]
![E524AB3F-A581-4648-A547-C283B26443FF](https://user-images.githubusercontent.com/1890926/99812252-8c57a480-2b89-11eb-9635-a7b84be0b441.png)

**Comment by mcsdevv** [comment]
Thanks for sharing @osdiab, I recommend creating a ticket so we can look further.

**Comment by osdiab** [comment, marked as answer]
Figured it out on my own. For all those who wonder:

The vercel command I used was this: `vercel --prod packages/website-next`. I specified `packages/website-next` because my project uses that as the Root Directory (while allowing for source files outside of it), since it's a monorepo. Turns out you're supposed to just run `vercel --prod` at the root of the project itself (not the root directory specified in project config), and the Vercel config recognizes automatically what the root directory is supposed to be.

I figured out in two ways:

1. I read [the source code of the `vercel` deploy script on github](https://github.com/vercel/vercel/blob/master/packages/now-cli/src/commands/deploy/latest.js#L211), then added some `console.log()` calls to the actual deploy script by editing the `node_modules/vercel/dist/index.js` file that gets created when you install Vercel. This revealed the values of the source path and the root directory values in the code, which looked suspicious to me since they both contained the root directory in them.
1. I also noticed that when I just ran `vercel --prod`, it would take forever - this was because I had some uncommitted large files in my local working tree, when I removed them the build happened as expected.

I learned from this that the `vercel` deploy command doesn't seem to read the git repo at all even if your project is configured with a Github source code integration, it instead uploads the source code on your machine to Vercel and runs the build commands directly on that. This matches up with the "branch" field in the deployments UI showing nothing for manual deploys.

It is unclear to me if it's really uploading the whole local folder or not, haven't delved into it but it seems to do that upload step much more quickly than I'd have expected if that were the case. Definitely more black magic I don't understand, would be nice if the Vercel CLI docs explained at least some of this instead of providing practically no details on what the deploy command does.

**Comment by interviewByte** [comment]
I am also trying deploy my angular application but after GitHub action run complete command when i visit to Vercel when i click to the link it redirect me to 404 page
with messgae:
 404: NOT_FOUND
Code: NOT_FOUND
ID: bom1::gl2mp-1718194968633-8a3897b8f312
